### PR TITLE
added servlet

### PR DESCRIPTION
**Pull request by lilymammadova**

### CODE REVIEW

File: src/test/java/org/liliya/hotelapp/controller/AppServletTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+        verify(request).setAttribute("error", "Price is required.");
+        verify(dispatcher).forward(request, response);
+    }
+
+    @Test
+    public void givenClientName_whenReserveApartment_thenMethodForwardsToSuccessJspPage() throws ServletException, IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+        RequestDispatcher dispatcher = mock(RequestDispatcher.class);
```
Comment (Alexmansar) [inline src/test/java/org/liliya/hotelapp/controller/AppServletTest.java]: You can create it as a class field, and just use in each test

File: src/main/java/org/liliya/hotelapp/controller/AppServlet.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import org.liliya.hotelapp.persistence.StatePersistence;
+import org.liliya.hotelapp.persistence.StatePersistenceImpl;
+import org.liliya.hotelapp.service.ApartmentService;
+import org.liliya.hotelapp.service.ApartmentServiceImpl;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+
+public class AppServlet extends HttpServlet {
```
Comment (Alexmansar) [inline src/main/java/org/liliya/hotelapp/controller/AppServlet.java]: In general good job)) but, can we sptil the logic? here we have a lof of responsibility 
maybe we can create different Servlets in general?

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@lilymammadova good job in general ) But I have small comments, please, check it)

File: src/main/java/org/liliya/hotelapp/controller/AppServlet.java (hunk 2 above)
Comment (lilymammadova) [reply src/main/java/org/liliya/hotelapp/controller/AppServlet.java]: created different servlets

File: src/test/java/org/liliya/hotelapp/controller/BaseServletTest.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import static org.mockito.Mockito.mock;
+
+public abstract class BaseServletTest {
+    protected HttpServletRequest request = mock(HttpServletRequest.class);
+    protected HttpServletResponse response = mock(HttpServletResponse.class);
+    protected RequestDispatcher dispatcher = mock(RequestDispatcher.class);
```
Comment (Alexmansar) [inline src/test/java/org/liliya/hotelapp/controller/BaseServletTest.java]: Could be just Mock annotation))

**Review by Alexmansar — APPROVED** [review]
@lilymammadova Good job)) please, fix 1 small comment and merge))

**Pull request merged**